manticore/platforms/linux.py

```diff
@@ -170,14 +170,25 @@ def tell(self):
         '''
         return self.pos
 
-    def seek(self, pos):
+    def seek(self, offset, whence = os.SEEK_SET):
         '''
         Returns the read/write file offset
         :rtype: int
         :return: the read/write file offset.
         '''
-        assert isinstance(pos, (int, long))
-        self.pos = pos
+        assert isinstance(offset, (int, long))
+        assert isinstance(whence, (int, long))
+
+        # horribly simplified (omits mode checks, error handling)
+        if whence == os.SEEK_SET:
+            self.pos = offset
+        elif whence == os.SEEK_CUR:
+            self.pos += offset
+        elif whence == os.SEEK_END:
+            self.max_size += offset
```

Comment:
What about a more complex version:

```Python
        new_position = 0
        if whence == os.SEEK_SET:
            new_position = offset
        elif whence == os.SEEK_CUR:
            new_position = self.pos + offset
        elif whence == os.SEEK_END:
            new_position = self.max_size + offset

        if new_position < 0:
            return -1

        if new_position < self.max_size:
            self.pos = new_position
        else:
            logger.warning("Symbolic seek: maximum file length reached")
            self.pos = self.max_size

        return self.pos
```

I'm not sure on how to handle the last case. I guess one can drop the last if/else. Technically there is no problem setting the position behind max_size. A read/write access should warn about it.